crates/evm/traces/src/identifier/local.rs

```diff
@@ -53,6 +54,19 @@ impl<'a> LocalTraceIdentifier<'a> {
             };
 
             if let Some(bytecode) = contract_bytecode {
+                let mut current_bytecode = current_bytecode;
+                if is_creation && current_bytecode.len() > bytecode.len() {
+                    // Try to decode ctor args with contract abi.
+                    if let Some(constructor) = contract.abi.constructor() {
+                        let constructor_args = &current_bytecode[bytecode.len()..];
+                        if constructor.abi_decode_input(constructor_args, false).is_ok() {
+                            // If we can decode args with current abi then remove args from
+                            // code to compare.
+                            current_bytecode = &current_bytecode[..bytecode.len()]
```

Comment:
this does assume that the metadata setting / output is the same for both cases but I think that is fair

Reply:
yeah, there's a diff score computed next 
```
   let score = bytecode_diff_score(bytecode, current_bytecode);
```

which should catch this

Reply:
@klkvr can you pls confirm above before merging? thank you!